packages/dds/sequence/src/test/intervalRebasing.spec.ts

```diff
@@ -480,4 +524,133 @@ describe("interval rebasing", () => {
+		containerRuntimeFactory.processAllMessages();
+		assertConsistent(clients);
+
+		assert.equal(clients[0].sharedString.getText(), "BA");
+
+		assertSequenceIntervals(
+			clients[0].sharedString,
+			clients[0].sharedString.getIntervalCollection("comments"),
+			[{ start: 1, end: 1 }],
+		);
+	});
+
+	it("changing interval to concurrently deleted segment detaches interval", () => {
+		// B-A
+		// ^
+		// (B)-A
+		//     ^
+		// (B)-(A)-C
+		//         ^
+		clients[0].sharedString.insertText(0, "A");
+		clients[2].sharedString.insertText(0, "B");
+		const collection_0 = clients[2].sharedString.getIntervalCollection("comments");
+		collection_0.add({ start: 0, end: 0, props: { intervalId: "0" } });
+		containerRuntimeFactory.processAllMessages();
+		assertConsistent(clients);
+		clients[1].sharedString.removeRange(0, 1);
+		clients[0].containerRuntime.connected = false;
+		containerRuntimeFactory.processAllMessages();
+		assertConsistent(clients);
+		clients[1].sharedString.removeRange(0, 1);
+		const collection_1 = clients[0].sharedString.getIntervalCollection("comments");
+		collection_1.change("0", { start: 0, end: 0 });
+		clients[2].sharedString.insertText(0, "C");
+		containerRuntimeFactory.processAllMessages();
+		assertConsistent(clients);
+		clients[0].containerRuntime.connected = true;
+		containerRuntimeFactory.processAllMessages();
+		assertConsistent(clients);
+
+		assert.equal(clients[0].sharedString.getText(), "C");
+
+		assertSequenceIntervals(
+			clients[0].sharedString,
+			clients[0].sharedString.getIntervalCollection("comments"),
+			[{ start: 0, end: 0 }],
+		);
+	});
+
+	it("changing interval endpoint while disconnected to segment also inserted while disconnected", () => {
+		// AC
+		// A-B-C
+		clients[0].sharedString.insertText(0, "AC");
+		containerRuntimeFactory.processAllMessages();
+		assertConsistent(clients);
+		const collection_0 = clients[0].sharedString.getIntervalCollection("comments");
+		collection_0.add({ start: 0, end: 0, props: { intervalId: "0" } });
+		clients[0].containerRuntime.connected = false;
+		clients[0].sharedString.insertText(1, "B");
+		collection_0.change("0", { start: 1, end: 1 });
+		clients[0].containerRuntime.connected = true;
+		containerRuntimeFactory.processAllMessages();
+		assertConsistent(clients);
+
+		assert.equal(clients[0].sharedString.getText(), "ABC");
+
+		assertSequenceIntervals(
+			clients[0].sharedString,
+			clients[0].sharedString.getIntervalCollection("comments"),
+			[{ start: 1, end: 1 }],
+		);
+	});
+
+	it("delete and insert text into range containing interval while disconnected", async () => {
+		// 012
+		// (0)-x-12
+		const intervals = clients[0].sharedString.getIntervalCollection("comments");
+		clients[0].sharedString.insertText(0, "012");
+		intervals.add({ start: 0, end: 2, props: { intervalId: "0" } });
+		assertSequenceIntervals(clients[0].sharedString, intervals, [{ start: 0, end: 2 }]);
+
+		clients[0].containerRuntime.connected = false;
```

Comment:
in general with the way the mocks work, this should function equivalently to if you move it to the start of the test (disconnecting a client will cancel all their in-flight ops in the mock implementation), which I generally find to be less confusing to reason about: as written it looks like there might be a difference between the interval addition and the insert of "x", but there isn't really.